solutions/USACO feb19-perimeter-silver.mdx

```diff
@@ -0,0 +1,119 @@
+---
+id: usaco-895
+source: USACO Silver 2019 January
+title: Icy Perimeter
+author: Alec Dewulf 
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_perimeter_silver_jan19.html)
+
+This problem can be solved using [Flood Fill](/silver/flood fill) (a Silver topic).
+
+## Implementation
+
+<LanguageSection>
+<CPPSection>
+
+
+```cpp
+
+#include <bits/stdc++.h>
+
+#define check(x) cout<<(#x)<<": "<<x<<" " << endl;
+#define io ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);
+#define ss second
+#define ff first
+#define pb push_back
+#define lb lower_bound
+#define ub upper_bound
+#define ld long double
+#define all(c) (c).begin(), (c).end()
+#define FOR(i, a, b) for (int i = a; i < b; i++)
+#define mp make_pair
+#define line cout << "--------------" << endl;
+
+typedef long long ll;
+typedef unsigned long long ull;
+
+int pct(int x) { return __builtin_popcount(x); }
+
+using namespace std;
+
+ifstream fin("perimeter.in");
+ofstream fout("perimeter.out");
+
+
+int n;
+char grid[1001][1001];
+bool visit[1001][1001];
+
+int a, p;
+// flood fill that updates perimeter and area for 
+// each new block
+void ff (int r, int c){
+    if (r >= n || c >= n || r < 0 || c < 0 || grid[r][c] == '.' )
+        return;
+
+    if (visit[r][c]){
+        return;
+    }
+    a ++;
+    visit[r][c] = true;
+    int co = 0;
+    if (r + 1 < n && grid[r + 1][c] == '.' || r + 1 >= n)
+         p++;
+    if (r -1 >= 0 && grid[r -1][c] == '.' || r - 1 < 0)
+        p ++;;
+    if (c + 1 < n && grid[r][c + 1] == '.' || c + 1 >= n)
```

Comment:
put parentheses around your conditions so people aren't confused by the order of operations...